packages/vite/src/node/preview.ts

```diff
@@ -41,33 +41,48 @@ export async function preview(
   )
 
   const options = config.server || {}
-  const hostname = options.host || 'localhost'
+  let hostname: string
+  if (options.host === undefined) {
```

Comment:
```suggestion
  if (options.host === undefined || options.host === 'localhost') {
```

Should also normalize `localhost` to `127.0.0.1` as well, otherwise the check on L63 will think  `vite --host localhost` as open.